api/types/swarm/secret.go

```diff
@@ -0,0 +1,30 @@
+package swarm
+
+// Secret represents a secret.
+type Secret struct {
+	ID string
+	Meta
+	Spec       *SecretSpec `json:",omitempty"`
+	Digest     string      `json:",omitempty"`
+	SecretSize int64       `json:",omitempty"`
+}
+
+type SecretSpec struct {
+	Annotations
+	Data []byte `json:",omitempty"`
```

Comment:
Again, probably don't want to omit the `Data` field if the secret is zero-length.